client/src/templates/Challenges/redux/index.js

```diff
@@ -231,13 +231,9 @@ export const reducer = handleActions(
         );
       return {
         ...state,
-        challengeFiles: [
-          ...state.challengeFiles.filter(x => x.fileKey !== fileKey),
-          {
-            ...state.challengeFiles.find(x => x.fileKey === fileKey),
-            ...updates
-          }
-        ]
+        challengeFiles: state.challengeFiles.map(file =>
+          file.fileKey === fileKey ? { ...file, ...updates } : { ...file }
+        )
```

Comment:
```suggestion
        challengeFiles: state.challengeFiles.map(challengeFile =>
          challengeFile.fileKey === fileKey ? { ...challengeFile, ...updates } : { ...challengeFile }
        )
```

Reply:
This is just because when I search `file`, I hope to only find the legacy references to what the API calls `file`.